pynest/nest/lib/hl_api_connection_helpers.py

```diff
@@ -236,3 +248,61 @@ def fixdict(d):
     projections = fixdict(projections)
     sps(projections)
     sr('ConnectLayers')
+
+
+def _check_input_nodes(pre, post, conn_spec):
+    '''
+    Check the properties of `pre` and `post` nodes:
+
+    * if `conn_spec` is "one_to_one", skip uniqueness check for better speed
+      and use data_connect
+    * if both `pre` and `post` are NodeCollections or can be converted to
+      NodeCollections (i.e. contain unique IDs), then proceed to "normal"
+      connect (potentially after conversion to NodeCollection)
+    * if both `pre` and `post` are arrays and contain non-unique items, then
+      we proceed to "data_connect"
+    * if at least one of them has non-unique items and they have different
+      sizes, then raise an error.
```

Comment:
Only this bullet point has a period. I suggest you use periods and capital first letter for all bullet points.